lms/templates/ccx/schedule.html

```diff
@@ -29,80 +29,88 @@
   <div id="new-ccx-schedule"></div>
 </div>
 
-<section id="enter-date-modal" class="modal" aria-hidden="true">
-  <div class="inner-wrapper" role="dialog">
-    <button class="close-modal">
-      <i class="icon fa fa-remove"></i>
+<section id="enter-date-modal" class="modal"
+         tabindex="1" role="dialog" aria-labelledby="ccx_schedule_set_date_heading">
+  <div class="inner-wrapper">
+    <button class="close-modal" tabindex="2">
+      <i class="icon fa fa-remove" aria-hidden="true"></i>
       <span class="sr">
         ${_("Close")}
       </span>
     </button>
     <header>
-      <h2></h2>
+      <h2 tabindex="2" id="ccx_schedule_set_date_heading"></h2>
     </header>
     <form role="form">
       <div class="field datepair">
-        <label></label>
-        <input placeholder="Date" class="date" type="text" name="date"/ size="11">
-        <input placeholder="Time" class="time" type="text" name="time"/ size="6">
+        <input placeholder="Date" class="date" type="text" name="date"
+               id="date" size="11" tabindex="2" aria-label="${_('Enter date format yyyy-mm-dd')}">
+        <input placeholder="Time" class="time" type="text" name="time" id="time" size="6"
+               tabindex="2" aria-label="${_('Enter time format hh-mm')}">
       </div>
       <div class="field">
-        <button type="submit" class="btn btn-primary">${_('Set date')}</button>
+        <button type="submit" class="btn btn-primary" tabindex="2">${_('Set date')}</button>
       </div>
     </form>
   </div>
 </section>
 
 <div class="ccx-schedule-sidebar">
-  <div class="ccx-sidebar-panel" id="dirty-schedule">
-    <h2>${_('Save changes')}</h2>
-    <form role="form">
-      <p>${_("You have unsaved changes.")}</p>
+  <div class="ccx-sidebar-panel" id="dirty-schedule" tabindex="1" role="group"
+       aria-labelledby="ccx_schedule_save_changes_heading">
+    <h2 id="ccx_schedule_save_changes_heading">${_('Save changes')}</h2>
+    <form>
+      <p id="message_save" tabindex="2">${_("You have unsaved changes.")}</p>
```

Comment:
tabindex is inappropriate here.  forcing a paragraph to be focusable would be confusing to a screen reader user as they would expect it to *do* something, like a link, button, or form control, which are natively focusable.  If a screen reader wants to read this, they would use arrow keys, not tab.